lib/PCP-server-Ruby-SDK/api/base_api_client.rb

```diff
@@ -0,0 +1,105 @@
+require 'net/http'
+require 'json'
+require 'uri'
+
+require_relative '../communicator_configuration'
+require_relative '../request_header_generator'
+require_relative '../errors/api_error_response_exception'
+require_relative '../errors/api_response_retrieval_exception'
+
+MERCHANT_ID_REQUIRED_ERROR = 'Merchant ID is required'
+COMMERCE_CASE_ID_REQUIRED_ERROR = 'Commerce Case ID is required'
+CHECKOUT_ID_REQUIRED_ERROR = 'Checkout ID is required'
+
+class BaseApiClient
+  def initialize(config)
+    @config = config
+    @request_header_generator = RequestHeaderGenerator.new(config)
+  end
+
+protected
+
+  def get_request_header_generator
+    @request_header_generator
+  end
+
+  def get_config
+    @config
+  end
+
+  def make_api_call(url, request_init, &parse_body)
+    parse_body ||= BaseApiClient.method(:parse_json)
+
+    uri = URI.parse(url)
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.use_ssl = uri.scheme == 'https'
+    
+    headers = @request_header_generator.generate_additional_request_headers(url, request_init || {})
```

Comment:
In den anderen SDKs wird hier der request modifiziert und nicht die header separat zurückgegeben. Vielleicht können wir das ja auch noch angleichen hier

Reply:
Yes for this we have to update the method as well as it currently updates the parameter. We have to look into how ruby does copying and have to make sure that it actually deals with a copy and not a modified source object. That might cause some issues later on